csharp/test/Microsoft.ML.OnnxRuntime.Tests.NetCoreApp/TrainingTest.cs

```diff
@@ -0,0 +1,287 @@
+using System.IO;
+using System.Linq;
+using Xunit;
+using Xunit.Abstractions;
+
+// This runs in a separate package built from EndToEndTests
+// and for this reason it can not refer to non-public members
+// of Onnxruntime package
+namespace Microsoft.ML.OnnxRuntime.Tests
+{
+    public partial class TrainingTest
+    {
+        private readonly ITestOutputHelper output;
+
+        public TrainingTest(ITestOutputHelper o)
+        {
+            this.output = o;
+        }
+
+        [Fact(DisplayName = "TestLoadCheckpoint")]
+        public void TestLoadCheckpoint()
+        {
+            using (var opt = new CheckpointState())
+            {
+                Assert.NotNull(opt);
+                string path = Path.Combine(Directory.GetCurrentDirectory(), "checkpoint.ckpt");
+                opt.LoadCheckpoint(path);
+            }
+        }
+
+        [Fact(DisplayName = "TestCreateTrainingSession")]
+        public void TestCreateTrainingSession()
+        {
+            using (var state = new CheckpointState())
+            {
+                Assert.NotNull(state);
+                string checkpointPath = Path.Combine(Directory.GetCurrentDirectory(), "checkpoint.ckpt");
+                state.LoadCheckpoint(checkpointPath);
+                string trainingPath = Path.Combine(Directory.GetCurrentDirectory(), "training_model.onnx");
+                string evalPath = Path.Combine(Directory.GetCurrentDirectory(), "training_model.onnx");
+                string optimizerPath = Path.Combine(Directory.GetCurrentDirectory(), "adamw.onnx");
+
+
+                //var trainingSession = new InferenceSession(trainingPath);
+                //var trainingSession = new TrainingSession(state, trainingPath, evalPath, "");
+                var trainingSession = new TrainingSession(state, trainingPath);
+
+                trainingSession.Dispose();
+            }
+        }
+
+        [Fact(DisplayName = "TestTrainingSessionTrainStep")]
+        public void TestTrainingSessionTrainStep()
+        {
+            using (var state = new CheckpointState())
+            {
+                Assert.NotNull(state);
+                string checkpointPath = Path.Combine(Directory.GetCurrentDirectory(), "checkpoint.ckpt");
+                state.LoadCheckpoint(checkpointPath);
+                string trainingPath = Path.Combine(Directory.GetCurrentDirectory(), "training_model.onnx");
+
+                var trainingSession = new TrainingSession(state, trainingPath);
+
+                float[] expectedOutput = TestDataLoader.LoadTensorFromFile("loss_1.out");
+                var expectedOutputDimensions = new int[] { 1 };
+                float[] input = TestDataLoader.LoadTensorFromFile("input-0.in");
+                Int32[] labels = { 1, 1 };
+
+                // Run inference with pinned inputs and pinned outputs
+                using (DisposableListTest<FixedBufferOnnxValue> pinnedInputs = new DisposableListTest<FixedBufferOnnxValue>(),
```

Comment:
>DisposableListTest<FixedBufferOnnxValue> pinnedInputs [](http://example.com/codeflow?start=23&length=53)

I think creating a single DisposableList<IDisposable> would be enough for all of that.

Reply:
Follows InferenceSession tests